### PR TITLE
upgrade lib version and add a fake service to the dockerfile

### PR DESCRIPTION
**Pull request by RazorSh4rk**
After hashicorp seemingly removed their local consul image the instfs build was breaking
Created a dummy server that responds the same as a consul kv store
https://github.com/RazorSh4rk/fakeconsul
https://developer.hashicorp.com/consul/api-docs/kv
Bumped consul lib version, that needed a minimum of node14

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ trap cleanup INT TERM EXIT
 set -e
 
 echo ": Build and start containers..."
-export COMPOSE_FILE=docker-compose.test.yml
+export COMPOSE_FILE=docker-compose.yml
 docker-compose build
 
 # docker-compose on mac doesn't flush stdout and stderr properly without a tty,
```
Comment (KarolyBujtor) [inline build.sh:19]: What do you think about the use just a `docker compose run` command with a shell script with a `depends_on` line in the `docker-compose.yml`, which encapsulates all of these lines below in a more readable format?

With this way there is no need to allocate tty.

File: docker-compose.yml [hunk 2]
```diff
@@ -2,23 +2,22 @@ version: '2'
...
+      # - node_modules:/usr/src/app/node_modules
     links:
       - consul
     environment:
       NODE_ENV: development
       CONSUL_HOST: "consul"
       CONSUL_PORT: "8500"
-      CONSUL_JWT_SECRET_PREFIX: "config/development/secrets"
-
+      CONSUL_JWT_SECRET_PREFIX: "/config/development/secrets"
```
Comment (KarolyBujtor) [inline docker-compose.yml:15]: I would add a `depends_on` line to this config to ensure the consule service is started before the app.

**Review by KarolyBujtor — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM instructure/node:12
+FROM instructure/node:14
```
Comment (kercsoj) [inline Dockerfile:1]: Node 14 was EOL 2023-04-30. Why not upgrading to Node 20 at least? There is an instructure/node:20 already available.

**Review by kercsoj — CHANGES_REQUESTED** [review]
I would use a newer Node version.

**Pull request merged**